src/Sentry.Profiling/SamplingTransactionProfilerFactory.cs

```diff
@@ -41,10 +45,24 @@ private SamplingTransactionProfilerFactory(SentryOptions options, SampleProfiler
         // Start a profiler if one wasn't running yet.
         if (Interlocked.Exchange(ref _inProgress, TRUE) == FALSE)
         {
+            if (!_sessionTask.IsCompleted)
+            {
+                _options.LogWarning("Cannot start a sampling profiler, the session hasn't started yet.");
+                _inProgress = FALSE;
+                return null;
+            }
+
+            if (!_sessionTask.IsCompletedSuccessfully)
```

Comment:
In these scenarios there's no profile associated with a transaction and so `TransactionProfiler?.Finish();` won't be called when the Transaction finishes right?
https://github.com/getsentry/sentry-dotnet/blob/175623c91c7c7a660ed3a5bafb6d4bd31ba86cb7/src/Sentry/TransactionTracer.cs#L368

What does happen to the profile though? Does it get cleaned up?

Reply:
> What does happen to the profile though? Does it get cleaned up?

There's no profile being collected if we don't create an instance of `SamplingTransactionProfiler`. Yes, the session is running in the background but it drops all samples unless there's a profiler..

> I'm thinking particularly since some of the stuff that happens during initialization can "block indefinitely"...

That's OK, we just won't collect any profiles.

>Is this the answer?  ` // TODO cancellation token based on Hub being closed? `

Nope, that's another improvement we could do in case a transaction is left unfinished while a hub is being closed. In that case, we don't really care about continuing the profiler (even until up to the given 30 second timeout) and we could stop it immediately. This isn't handled by the hub at the moment, though.